VultureProject/darwin
# Version 1.5

# Changelog

## Core


## Filters

### BUFR
- Add 'float' type to valid input formats
- [BREAKING] Buffer checks keys in Redis, won't start if key is of wrong type and will delete the key to avoid using stale/improper data
- [NEW] Add 'sum' buffer
  - Increments a redis key to calculate sum over a time span
  - Can take positive and negative values, represented as integers or floats (handled as floats internally)
  - 'required_log_lines' represent a limit under which not to use the absolute rounded value of the sum
  - Value is never reinserted in case of error -> won't stack sums in different time intervals
  - [FIX] Removing the modification of the wrong statistic value.

## Manager
- [FIX] Avoid Manager crashing when updating a configuration with syntax errors
  - Added associated tests
- Add RFC3339 'timestamp' field to stats generated

## CMakeLists
- Add VAST file inclusion (proprietary)
- Add VAML file inclusion (proprietary)
- Add option to activate coverage flags

## Functional tests
- Add mock test file
- Change tests to include new 'float' input type in entries
- Update anomaly test to adapt to new algorithm
- Add tests for new 'sum' filter
  - test with one value sent
  - test with several (negative, positive, integer) values to test correct summing
  - test with sum under 'required_log_lines'
  - test with existing correct type key overrode at startup
  - test with existing incorrect type key, preventing filter to start

## Test framework
- [FEATURE] Add capability to check lines are present in logs
- [FEATURE] Add an ApproxDict class to check float equality with delta in dictionaries
- [FEATURE] Add a generic filter function to validate fields in generated filters' alerts
- Reuse MOCK_PATH into TEST_FILES_DIR to specify runtime temp dir
- Improve Valgrind print returns
- Remove useless sleep after stop
- Add mock test file for VAML/VAST

## TEST filter
- Allow ftest to get lists AND simple values
- Simple values are interpreted as string, but can also be numbers

## Dockerfile
- Build boost:date_time (required for proprietary VAML)
